src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -169,18 +169,24 @@ AssemblyDefinition ResolveRuntimeAssemblyForReferenceAssembly (LockFile lockFile
 				LogWarning ($"Could not resolve target for '{TargetMoniker}'");
 				return null;
 			}
-			var libraryPath = lockFile.Libraries.FirstOrDefault (x => x.Name == assemblyNameDefinition.Name);
-			if (libraryPath == null)
-				return null;
-			var library = target.Libraries.FirstOrDefault (x => x.Name == assemblyNameDefinition.Name);
-			if (library == null)
-				return null;
-			var runtime = library.RuntimeAssemblies.FirstOrDefault ();
-			if (runtime == null)
-				return null;
-			var path = Path.Combine (NuGetPackageRoot, libraryPath.Path, runtime.Path);
-			LogDebugMessage ($"Attempting to load {path}");
-			return resolver.Load (path, forceLoad: true);
+			foreach (var folder in lockFile.PackageFolders) {
+				var path = assemblyPath.Replace (folder.Path, string.Empty);
+				var libraryPath = lockFile.Libraries.FirstOrDefault (x => path.StartsWith (x.Path));
+				if (libraryPath == null)
+					continue;
+				var library = target.Libraries.FirstOrDefault (x => x.Name == libraryPath.Name);
```

Comment:
I recommend using `String.Compare (x.Name, libraryPath.Name, StringComparison.OrdinalIgnoreCase) == 0` instead of the `==` operator here